subprojects/docs/src/docs/release/notes.md

```diff
@@ -215,6 +215,15 @@ See the user guide section on the [`plugins {}` block](userguide/plugins.html#se
 ### Example new and noteworthy
 -->
 
+### Experimental caching for C/C++ compilation
+
+This release adds support for using the build cache with C/C++ compilation.
+To enable that feature the system property `org.gradle.native.caching` has to be set to `true`.
+Caching C/C++ compilation is still experimental and some caveats apply:
+
+* The toolchain is not tracked as an input to the compilation tasks
+* System headers provided by the compiler are not tracked
+
```

Comment:
I think we need to explain what the impact on the user is here. Something like this:

* the compiler type and version are not tracked as inputs, thus object files produced by different compilers can be shared
* system headers are not tracked as inputs, therefore object files produced with different versions of system libraries can be shared

"can be shared" is probably not the right phrasing, though